llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -1377,6 +1378,20 @@ void SelectionDAGLegalize::LegalizeOp(SDNode *Node) {
   }
 }
 
+// Helper function that generates an MMO that considers the alignment of the
+// stack, and the size of the stack object
+static MachineMemOperand *getStackAlignedMMO(SDValue StackPtr,
+                                             MachineFunction &MF) {
+  auto &MFI = MF.getFrameInfo();
+  int FI = cast<FrameIndexSDNode>(StackPtr.getNode())->getIndex();
```

Comment:
.getNode() is unnecessary

Reply:
`int FI = cast<FrameIndexSDNode>(StackPtr)->getIndex();`

Reply:
Done!